src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -1943,16 +1947,18 @@ private DisposablePluginService SearchAndInitializeProjectCachePluginFromGraph(P
                 return new DisposablePluginService(null);
             }
 
-            ErrorUtilities.VerifyThrowInvalidOperation(
-                cacheItems.Count == 1,
-                "OnlyOneCachePluginMustBeSpecified",
-                string.Join("; ", cacheItems.Select(ci => ci.PluginPath)));
+            if (cacheItems.Count != 1)
```

Comment:
Could be, but it may end up looking uglier since there's specific details like the exception type and what specific args to send to its particular constructor.

Reply:
I was considering commenting similar, maybe the next iteration creates some plugin-specific ErrorUtilities method.

Reply:
The exception creation patterns are so particular to ProjectCacheException I'd rather just have factory methods on the type. We could pass in lambdas to ErrorUtilities but I think that would just complicate it without much benefit.